src/app/backend/db/service.ts

```diff
@@ -1,13 +1,27 @@
-import { internal_getDbClient } from "./config";
+import { dontUseOutsideConfigOrTests_getDbClient } from "./config";
 import { camelCaseDbColumnNames } from "./decorators";
-import { Pool } from "pg";
+import { Pool, PoolClient } from "pg";
 
 export class DbService {
-  private dbClient: Pool = internal_getDbClient();
+  private dbPool: Pool = dontUseOutsideConfigOrTests_getDbClient();
+  private dbClient: PoolClient | undefined;
 
   @camelCaseDbColumnNames
   async query(querySql: string, values?: unknown[]) {
-    return await this.dbClient.query(querySql, values);
+    return this.dbClient
+      ? await this.dbClient.query(querySql, values)
+      : await this.dbPool.query(querySql, values);
+  }
+
+  async connect() {
```

Comment:
@nickclyde this is the new config file where we're exposing individual pool clients to make the transaction code works